examples/python-cicd-with-cloudbuilder/README.md

```diff
@@ -0,0 +1,135 @@
+# Basic Python CI With Cloud Source Repositories and Cloud Build
+
+## Overview
+This repo contains code and instructions for using Google Cloud Source Repositories and Cloud Build to execute unit tests upon code check-in.
+
+The basic steps we need to follow are:
+1. Create a new project in Cloud Source Repository and clone it to your machine
+2. Create your code and unit tests
+3. Create a Python Cloud Builder (more below)
+4. Create a cloud build trigger (more below)
+5. Create a cloudbuild.yaml file that executes your tests using your python cloud builder.
+
+## 1. Create a new project in Cloud Source Repository
+* Go to https://source.cloud.google.com/
+* Click 'Add repository'
+* Choose 'Create new repository'
+* Specify a name, and your project name
+* Follow the instructions to 'git clone' the empty repo to your workstation and use it like you would a git repo
+
+## 2. Create your code and unit tests
+Creating unit tests is beyond the scope of this README, but if you review the tests in tests/ you'll quickly get the idea. 
+I'm using pytest as the testing suite for this project. Before proceeding make sure you can run your tests from the 
+command line
+
+```
+python -m pytest 
+
+```
+
+Or if you want to be fancy and use the coverage plug-in
+
+```
+python -m pytest --cov=my_module tests/
+
+```
+
+If everything goes well you should expect to see output like this, showing successful tests:
+
+```
+$ python -m pytest --cov=my_module tests/
+============================================================================= test session starts ==============================================================================
+platform darwin -- Python 3.7.3, pytest-4.6.2, py-1.8.0, pluggy-0.12.0
+rootdir: /Users/mikebernico/dev/basic-cicd-cloudbuild
+plugins: cov-2.7.1
+collected 6 items
+
+tests/test_my_module.py ......                                                                                                                                           [100%]
+
+---------- coverage: platform darwin, python 3.7.3-final-0 -----------
+Name                     Stmts   Miss  Cover
+--------------------------------------------
+my_module/__init__.py        1      0   100%
+my_module/my_module.py       4      0   100%
+--------------------------------------------
+TOTAL                        5      0   100%
+
+
+```
+
+Now that our tests are working locally, we want to configure GCP to run them every time we push new code.
+
+## 3. Building a Python Cloud Builder
+In order to execute python based tests in cloud builder you need a python cloud builder. 
+A cloud builder is just a docker container with the stuff you need in it. Google doesn't distribute a 
+python cloud builder, so we have to build one ourselves. I've created one in '/python-cloud-builder.' All you need to do 
+is run this:
+
+```
+cd python-cloud-builder
+gcloud builds submit --config=cloudbuild.yaml .
```

Comment:
You may want to make it clear in the intro to the readme that they need the gloud SDK installed. Alternatively (and preferably) you can make the example around cloud shell which makes things easier.

Assume very little of your users, they may be new to GCP and CSR and everything else. Breadcrumbs or links to other docs can help a lot. Not everyone agrees with me on this so you can take or leave it.

Reply:
> Assume very little of your users,
And people in general.  :)
Good advice, added.